services/api/src/api/routes/rows.py

```diff
@@ -67,19 +68,27 @@ class ParquetDataProcessingError(Exception):
 
 # TODO: how to invalidate the cache when the parquet branch is created or deleted?
 @lru_cache(maxsize=128)
-def get_parquet_fs(dataset: str, hf_token: Optional[str]) -> HfFileSystem:
-    """Get the parquet filesystem for a dataset.
-
-    The parquet files are stored in a separate branch of the dataset repository (see PARQUET_REVISION)
+def get_hf_fs(hf_token: Optional[str]) -> HfFileSystem:
+    """Get the Hugging Face filesystem.
 
     Args:
-        dataset (str): The dataset name.
         hf_token (Optional[str]): The token to access the filesystem.
+    Returns:
+        HfFileSystem: The Hugging Face filesystem.
+    """
+    return HfFileSystem(token=hf_token)
 
+
+def get_hf_parquet_uris(paths: List[str], dataset: str) -> List[str]:
+    """Get the Hugging Face URIs from the Parquet branch of the dataset repository (see PARQUET_REVISION).
+
+    Args:
+        paths (List[str]): List of paths.
+        dataset (str): The dataset name.
     Returns:
-        HfFileSystem: The parquet filesystem.
+        List[str]: List of Parquet URIs.
     """
-    return HfFileSystem(dataset, repo_type="dataset", revision=PARQUET_REVISION, token=hf_token)
+    return [f"hf://datasets/{dataset}@{quote_plus(PARQUET_REVISION)}/{path}" for path in paths]
```

Comment:
OK, it's hard to forge an URL (quote_plus). Shouldn't we have it in huggingface_hub instead? cc @Wauplin

Reply:
We have `save_quote` in `huggingface_hub.hf_file_system`.

Reply:
Otherwise, we can use `HfFileSystemResolvedPath`.unresolve()`. Let me do the change.

Reply:
OK, but I meant: huggingface_hub could provide a way to forge the URLs

Reply:
OK :)  Note that I didn't ask you to change the PR in any way, it's perfect for now. It was a feature request for hfh